.github/actions/compile-commit/action.yml

```diff
@@ -7,9 +7,24 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: Get the hash of the tree at HEAD
+      id: repo-hash
+      shell: bash
+      run: |
+        TREE_HASH=$( git rev-parse HEAD: )
+        echo "tree-hash=${TREE_HASH}" >> $GITHUB_OUTPUT
+    - name: Cache compile-commit action success
+      id: check-compile-commit-success
+      uses: actions/cache@v3
+      with:
+        path: compile-commit-success.txt
+        # use tree hashes instead of commit hashes to reuse cache when changes in commits are reorganized without changing the workspace
+        key: compile-commit-success-${{ github.event.pull_request.head.repo.full_name }}-${{ steps.repo-hash.outputs.tree-hash }}
```

Comment:
`${{ github.event.pull_request.head.repo.full_name }}` - this renders out to a constant value `trino`. Does something need to be changed if we want to mention the branch name (or PR number) here (so that invalidating the cache for a specific PR/branch is easy to do if needed anytime).